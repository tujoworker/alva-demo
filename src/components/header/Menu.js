/**
 * Component
 *
 * @author Tobias HÃ¸egh <tobias@tujo.no>
 */

// Button Menu
// use also role="button"

// import './Menu.css'
// import { FormattedMessage } from 'react-intl'
import { NavLink } from 'alva/static'
import { inject } from 'mobx-react'
import PropTypes from 'prop-types'
import React, { Component } from 'react'
import styled from 'react-emotion'

@inject('locale')
export default class Menu extends Component {
  static propTypes = {
    locale: PropTypes.shape({
      L: PropTypes.func.isRequired
    }).isRequired
  }

  render() {
    return (
      <Nav>
        <NavLink exact to="/">
          {this.props.locale.L('menu.main.home')}
          {/* <FormattedMessage id="menu.main.blog" /> */}
        </NavLink>
        <NavLink to="/about">
          {this.props.locale.L('menu.main.about')}
        </NavLink>
        <NavLink to="/form">
          {this.props.locale.L('menu.main.form')}
        </NavLink>
        <NavLink to="/auth">
          {this.props.locale.L('menu.main.login')}
        </NavLink>
        <NavLink to="/dynamic">
          {this.props.locale.L('menu.main.dynamic')}
        </NavLink>
        <NavLink to="/blog">
          {this.props.locale.L('menu.main.blog')}
        </NavLink>
      </Nav>
    )
  }
}

const Nav = styled('nav')`
  ${'' /* font-size: 1.5em; */};
  ${'' /* white-space: nowrap; */} width: 100%;
  ${'' /* display: inline-block !important; */} background: #fff;

  a {
    padding: 0.8em;
    display: inline-block;
    text-decoration: none;
    font-style: normal;
    color: #000;
  }

  a.active {
    color: var(--main-bg-color);
  }
`
