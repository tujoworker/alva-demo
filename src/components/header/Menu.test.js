/**
 * A simple test, to test the results of components
 *
 * @author Tobias HÃ¸egh <tobias@tujo.no>
 */

//for older browsers
// import '../../core/startup/polyfill'

//test core packages
import { shallow } from 'alva/tdd'
import React from 'react'

import toJson from 'enzyme-to-json'

import Menu from './Menu'

// The API Docs
// http://airbnb.io/enzyme/docs/api/
// https://facebook.github.io/jest/docs/en/expect.html

// Attention: use .wrappedComponent once we @inject a component
// const store = fakeProps(__dirname, './Menu')
const store = {
  // locale: { L: () => 'x' } // because of @inject('locale')
  locale: { L: jest.fn() } // because of @inject('locale')
}

describe('Component renders correctly', () => {
  const MenuWrap = shallow(<Menu.wrappedComponent {...store} />)

  it('have to match snapshot', () => {
    expect(toJson(MenuWrap)).toMatchSnapshot()
  })

  it('has a href to / and exact attribute', () => {
    expect(MenuWrap.find('[to="/"][exact]').exists()).toBe(true)
  })
})
