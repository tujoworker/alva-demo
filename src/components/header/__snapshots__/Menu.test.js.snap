// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly have to match snapshot 1`] = `
<Styled(nav)>
  <SmartLink
    exact={true}
    to="/"
  />
  <SmartLink
    to="/about"
  />
  <SmartLink
    to="/form"
  />
  <SmartLink
    to="/auth"
  />
  <SmartLink
    to="/dynamic"
  />
  <SmartLink
    to="/blog"
  />
</Styled(nav)>
`;
